### ISSUE TITLE
[bug] tooltip should not have an `on` attribute

### ISSUE THREAD

**Issue by bennypowers** [RedHat-UX/red-hat-design-system#698]
### Describe the bug

#693 introduced a regression in tooltip where the context `on` state which had previously been private to shadow DOM was reflected to a host attribute. A number of PRs (#660, #661, #675 for example) took efforts to remove the public `on` attribute. The reason being that users should not be able to override the colour context via attribute, since it is meant to inherit from the parent context.

### Which Component?

rh-tooltip

### Steps to reproduce

1. Open the [tooltip demo](https://ux.redhat.com/components/tooltip/demo/)
2. in the js console, execute this snippet:
   ```js
   const darkTooltip = document.querySelector('[color-palette="darkest"] rh-tooltip');
   darkTooltip.setAttribute('on', 'light');
   darkTooltip.show(); 
   ```

### Expected behaviour

The tooltip should display dark text on a light background

### Screenshots

![image](https://user-images.githubusercontent.com/1466420/211754697-aa62e226-9a9c-45b2-aa2c-d12dfbdd2445.png)

### Element HTML

```HTML
<rh-tooltip on="light">
  <span slot="content">Should be dark theme</span>
</rh-tooltip>
```

### Operating System (OS)

Linux

### Additional Info

We can wrap the tooltip shadow DOM in a container element to set the private context, like we do in other element:
```ts
@colorContextConsumer() private on: ColorTheme = 'light';

render() {
  const { on } = this;

  return html`
    <div id="rhds-wrapper" class=${classMap({ [on]: !!on })}>
      ${super.render()}
    </div>
  `;
}
```
```css
/* shadow css */
#rhds-wrapper {
  display: contents;
}
```

**Comment by bennypowers** [comment]
Paired with @brianferry on this earlier. The issues stemmed from confusion on the `attribute` config property for `@colorContextConsumer(config)`, which *should* default to false, but currently defaults to `'on'`.

a subsequent PR should
1. change the default to false
2. document the config properties inline, so they appear in lsp completion menus, hover, etc.

**Comment by bennypowers** [comment]
closed by #702
